assets/core-capi/core-cluster-api.yaml

```diff
@@ -499,27 +459,25 @@ data:
         - clusterresourcesets
       sideEffects: None
   metadata: |
-    # maps release series of major.minor to cluster-api contract version
-    # the contract version may change between minor or major versions, but *not*
-    # between patch versions.
-    #
-    # update this file only when a new major or minor version is released
     apiVersion: clusterctl.cluster.x-k8s.io/v1alpha3
+    metadata:
+      creationTimestamp: null
     releaseSeries:
-      - major: 1
-        minor: 0
-        contract: v1beta1
-      - major: 0
-        minor: 4
-        contract: v1alpha4
-      - major: 0
-        minor: 3
-        contract: v1alpha3
+    - contract: v1beta1
+      major: 1
+    - contract: v1alpha4
+      minor: 4
+    - contract: v1alpha3
+      minor: 3
+    - contract: v1beta1
+      major: 99
+      minor: 99
 kind: ConfigMap
 metadata:
   creationTimestamp: null
   labels:
     provider.cluster.x-k8s.io/name: cluster-api
     provider.cluster.x-k8s.io/type: core
+    provider.cluster.x-k8s.io/version: "99.99"
```

Comment:
If there is a configmap that matches a selector, ClusterAPI operator will try to apply its content each reconciliation cycle. When CAPI is running on vanilla k8s it means that the operator doesn't let users delete components, for openshift we can use this feature to perform `apply` style upgrades because operator doesn't care if configmap was changed between reconciliation.

Reply:
Important note: CRDs, RBAC, and image changes are managed by CVO.